### PR TITLE
Issues with origin data

### LINKED ISSUE #2825: Individual: Origin Data - more informative information about Endothelial surface are and BSA
image

What is shown now are some internal descriptions, I think what should be shown is the selection that has been used in PK-Sim:

image

### LINKED ISSUE #2826: Individual: Origin Data - wrong unti for "Height"
Sent a simulation from PK-Sim with the following individual to MoBi:

image

Height is shown as 16 cm, which is wrong:

image

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #2826
Fixes #2825

# Description
There are some errors calculating display values for ExtendedParameters when mapping an individual to an individual building block. We need to do display unit conversions and modify the calculation method extended property display value

Also, core update included

## Type of change

Please mark relevant options with an `x` in the brackets.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):
image

# Questions (if appropriate):

### CODE REVIEW

File: src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs [hunk 1]
```diff
@@ -128,23 +134,31 @@ public override IndividualBuildingBlock MapFrom(Individual individual)
...
-         addOriginDataToBuildingBlock(buildingBlock, keyForOriginDataParameter(key, parameter), $"{parameter.Value} {parameter.Unit}");
+         var displayValue = originDataFormattedForDisplay(parameter);
+         addOriginDataToBuildingBlock(buildingBlock, keyForOriginDataParameter(key, parameter), $"{displayValue} {parameter.Unit}");
+      }
+
+      private string originDataFormattedForDisplay(OriginDataParameter parameter)
+      {
+         var dimension = _dimensionRepository.DimensionForUnit(parameter.Unit);
+         var unit = dimension.FindUnit(parameter.Unit);
+         return _formatter.Format(dimension.BaseUnitValueToUnitValue(unit, parameter.Value)).TrimEnd('0', '.');
```
Comment (rwmcintosh) [inline src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs:153]: Is this desirable, or is there another way? I want to remove ```xxx.00``` from displayed values.

File: src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs [hunk 2]
```diff
@@ -128,23 +134,31 @@ public override IndividualBuildingBlock MapFrom(Individual individual)
       private void addCalculationMethodOriginDataToBuildingBlock(IndividualBuildingBlock buildingBlock, CalculationMethod calculationMethod)
       {
          var repInfo = _representationInfoRepository.InfoFor(RepresentationObjectType.CATEGORY, calculationMethod.Category);
-         addOriginDataToBuildingBlock(buildingBlock, repInfo.DisplayName, calculationMethod.Category);
+         addOriginDataToBuildingBlock(buildingBlock, repInfo.DisplayName, calculationMethod.DisplayName);
```
Comment (rwmcintosh) [inline src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs:137]: Here's how we are changing the displayed value of calculation methods. This is the property displayed in PK-Sim

File: src/PKSim.Presentation/Presenters/Parameters/TableParameterPresenter.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ public interface ITableParameterPresenter : ICommandCollectorPresenter
       void Save();
       void AddPoint();
       TableFormula EditedFormula { get; }
-      IEnumerable<ValuePointDTO> AllPoints();
+      IEnumerable<ValuePointDTO> AllPoints { get; }
```
Comment (rwmcintosh) [inline src/PKSim.Presentation/Presenters/Parameters/TableParameterPresenter.cs:26]: I did a global text search/replace for AllPoints() => AllPoints and this one also got changed in the process. I suppose it is ok this way. LMK

File: src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs (hunk 2 above)
Comment (msevestre) [reply src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs:137]: agrred

File: src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs (hunk 1 above)
Comment (msevestre) [reply src/PKSim.Core/Mappers/IndividualToIndividualBuildingBlockMapper.cs:153]: that definitley feels like a hack lol
what does this do? TrimEnd?

File: src/PKSim.Presentation/Presenters/Parameters/TableParameterPresenter.cs (hunk 3 above)
Comment (msevestre) [reply src/PKSim.Presentation/Presenters/Parameters/TableParameterPresenter.cs:26]: Loooks good to me

**Review by msevestre — COMMENTED** [review]
Looks good to me.
Not sure why the hack with removing the 0s since we don't do it anywhere and it's a lost of precision. It's weird to display some numbers with 2 digits and some without. So unless this was requested (please link to issue so that I can see the rational), let's remove?

**Comment by msevestre** [comment]
image

those are input but the same goes for the output

**Pull request merged**